lib/mix/lib/mix/tasks/test.ex

```diff
@@ -14,17 +30,91 @@ defmodule Mix.Tasks.Test do
           Mix.raise("Failed to cover compile directory: " <> compile_path)
       end
 
-      output = opts[:output]
+      formatters = Keyword.get(opts, :formatters, @default_formatters)
```

Comment:
`:console` is not really a formatter because it doesn't print it for every module. It is just a summary. So I would rather have a `summary: true | false` option.

Reply:
According to Erlang `cover:analyse/2` docs it will print data for all cover compiled modules. However I can misunderstanding your comment.

Reply:
Do you mean that we have also added an option to print all of the results to the terminal? If so, that's a different feature than the one we were debating. Let's focus on the summary feature first. Other additions should be considered separately.

Reply:
That is the exact result of what this PR produces https://github.com/elixir-lang/elixir/pull/7664#issuecomment-387730487. It display all modules and their respective coverage percentage. You meant something else? Like we should display only overall coverage like:

```
Generating cover results ...

Coverage: 72.32%

Generated HTML coverage results in 'cover' directory
```

?

Reply:
Ok. Sorry for the misunderstanding. The current output is fine. However, I wouldn't call it a formatter because what we print to the console and what we write to html files are two very distinct things. My proposal is to remove the idea of formatters and always generate HTML as before. Let's call the new reporting that you have added as part of this PR "Summary" and it can be disabled via the `summary: true | false` option.